symengine/python/symengine/lib/symengine_wrapper.pyx

```diff
@@ -153,11 +143,23 @@ def sympify(a, raise_error=True):
         return RealDouble(a)
     elif isinstance(a, complex):
         return ComplexDouble(a)
-    else:
-        return sympy2symengine(a, raise_error)
+    elif isinstance(a, tuple):
+        v = []
+        for e in a:
+            v.append(sympify(e, True))
+        return tuple(v)
+    elif isinstance(a, list):
+        v = []
+        for e in a:
+            v.append(sympify(e, True))
+        return v
+    elif hasattr(a, '_symengine_'):
+        return a._symengine_()
+    elif hasattr(a, '_sympy_'):
+        return sympy2symengine(a._sympy_(), raise_error)
+    return sympy2symengine(a, raise_error)
 
 cdef class Basic(object):
-    cdef RCP[const symengine.Basic] thisptr
```

Comment:
Where did this `thisptr` go? I am surprised it's working without it, as I thought it must be declared, either here, or in every single subclass.